### DISCUSSION TITLE
Best way to trim whitespace from :pipe command output?

### DISCUSSION THREAD

**Discussion by michaeljones** [helix-editor/helix#5530] [Q&A] [answered]
I've just discovered a great pairing between the [pastel](https://github.com/sharkdp/pastel) command and helix where you can select a hex colour code (like #8a6d3b in a css file) and run `:pipe pastel format hsl` to convert the selected colour to HSL syntax instead.

Unfortunately the `pastel` command prints the output on a new line so there is trailing new line character which means:

```
.alert-warning {
  color: #8a6d3b;
}
```
becomes:
```
.alert-warning {
  color: hsl(38, 40.1%, 38.6%)
;
}
```
What is the best way to address that? My first thought was to have some kind of `:pipe-trim` command that would trim whitespace from the output. I guess I could also set up my own executable on my path called `pastel-trim` which called `pastel` and removed the whitespace from the output. Or maybe there is some simple key sequence that'll handle it though I'd rather it was a seamless operation without addition clean up key strokes required.

I'm also curious how best to go about creating keyboard shortcuts for such that it might be possible to have a menu of formats to convert to. But I'm not sure that that is possible and I haven't explored yet.

**Comment by lukehsiao** [comment, marked as answer]
This isn't a helix-command, but you can just use general pipes to get what you want.

Some options that work for your example would be:

```
:pipe pastel format hsl | xargs echo -n 
```

or

```
:pipe pastel format hsl | tr -d '\n'  <-- this remove ALL newlines though, so this assumes only a single line output
```
![newline](https://github.com/helix-editor/helix/assets/7573542/c3d245d5-426f-49da-9ced-4e6e8b9c2dab)

**Reply by kirawi** [reply]
Or https://git-scm.com/docs/git-stripspace

**Reply by lukehsiao** [reply]
No, probably not. From the docs you linked:

> add a missing \n to the last line if necessary.

This will _create_ the awkward newline before the semi-colon that OP was wanting to get rid of.

Demo:

<div align="center">

![newline](https://github.com/helix-editor/helix/assets/7573542/dab98000-a485-42b6-b6fd-3cd761cb634f)

</div>
